gammapy/maps/tests/test_wcs.py

```diff
@@ -121,3 +122,25 @@ def test_wcsgeom_contains(npix, binsz, coordsys, proj, skydir, axes):
     if not geom.is_allsky:
         coords = [0.0, 0.0] + [ax.center[0] for ax in geom.axes]
         assert_allclose(geom.contains(coords), np.zeros((1,), dtype=bool))
+
+def test_wcsgeom_solid_angle():
+    binsz=1.0
+    npix=100
+
+    # create dummy axes
+    axis1 = MapAxis.from_edges(np.arange(0,4))
+    axis2 = MapAxis.from_edges(np.arange(10,20))
+    # Test using cartesian geometry
+    geom = WcsGeom.create(skydir=(0, 0), npix=(npix, npix), binsz=binsz, coordsys='GAL',proj='CAR',axes=[axis1,axis2])
+
+    solid_array = geom.solid_angle()
+
+    # Check array size
+    assert solid_array.shape == (9,3,100,100)
+
+    # Test bin size at b=0
+    solid_lat0 = binsz*np.pi/180 * (np.sin(binsz * np.pi / 180.)) * u.sr
```

Comment:
I'm -1 to have such code, that re-implements the CAR projection, in our tests. If you look around in Gammapy, you will find 1000s of lines of such complex tests, that are hard to understand. In my experience, this is more of a problem than an asset.

I would just hard-code the reference values of the pixels in the test.
There could still be a comment saying if the reference values just establish the current result, or if they can be understood. In this case, I think a value of very roughly `binsz ** 2` is expected, so that could be mentioned in a comment.

One more thing that might be worth testing is what happens for pixels "outside the image", e.g. at the edge for AIT projection. Ideally the solid angle function would return NaN there and not error out. For partially outside pixels with e.g. one corner outside, ideally we'd get NaN also or some useful value, and not e.g. a crazy large or small value because the algorithm fails and then that might cause problems with analysis in those cases.

Reply:
I beg to differ. Giving directly the numerical value does not give any information on how it was extracted. 

OK for the test with AIT external pixels.

Reply:
In case you haven't seen it: when we implemented SkyImage, we already spent quite some time trying to get useful test cases and to validate the coordinates and solid angle methods:
https://github.com/gammapy/gammapy/blob/28dc9ac7ffad8dc1e2f56e600302ce9a07021142/gammapy/image/tests/test_core.py#L20
My suggestion would have been to simply copy and adapt those to the new Map API, instead of trying to develop something new.
Although by now, I have changed my opinion a bit on how tests should be structured, preferring as simple and concrete tests over e.g. the way you're writing the tests here now (adding lines that re-implement CAR projection) or like we did in many other places in Gammapy, e.g. the complex way the test parametrisation was done via `test_image` for `SkyImage`; I already find that hard to read and maintain and would prefer a little bit of code duplication and having the reference values directly in the test function, i.e. to have two or three test functions each with one test case here.